lib/media/track/remotemediatrack.js

```diff
@@ -210,20 +211,26 @@ function shimMediaElement(mediaElement) {
 }
 
 function playIfPausedWhileInBackground(remoteMediaTrack) {
-  const { _log: log } = remoteMediaTrack;
+  const { _log: log, kind } = remoteMediaTrack;
   function onVisibilityChanged() {
     if (document.visibilityState === 'visible') {
       // NOTE(mpatwardhan): when document regains foreground restore elements that were
-      // unintentionally paused.
-      [...remoteMediaTrack._attachments.values()].forEach(el => {
-        const shim = shimmedElements.get(el);
-        if (el.paused && shim && !shim.isPausedIntentionally()) {
-          el.play().then(() => {
-            log.info('successfully played unintentionally paused element');
-          }).catch(err => {
-            log.warn('error playing paused element', err);
-          });
-        }
+      // unintentionally paused. For audio elements delay 2 seconds to ensure that
+      // if local audio track was being re-acquired it had chance to finish acquiring.
+      const delay = kind === 'audio' ? 2000 : 1;
+      const delayPromise = waitForSometime(delay);
+      delayPromise.then(() => {
+        [...remoteMediaTrack._attachments.values()].forEach(el => {
+          const shim = shimmedElements.get(el);
+          if (el.paused && shim && !shim.isPausedIntentionally()) {
+            log.info(`playing paused ${kind} element`);
+            el.play().then(() => {
+              log.info(`successfully played unintentionally paused ${kind} element`);
+            }).catch(err => {
+              log.warn(`error playing paused ${kind} element`, err);
+            });
+          }
+        });
```

Comment:
Let's play it safe and apply the delay for video as well:
```suggestion
      // NOTE(mpatwardhan): On iOS Safari, play inadvertently paused HTMLAudioElements with
      // a delay to make sure any inadvertently ended LocalMediaTracks are restarted first. Otherwise,
      // play() will reject with a NotAllowedError.
      waitForSomeTime(2000).then(() => {
        remoteMediaTrack._attachments.forEach(el => {
          const shim = shimmedElements.get(el);
          if (el.paused && shim && !shim.isPausedIntentionally()) {
            log.info(`playing paused ${kind} element`);
            el.play().then(() => {
              log.info(`successfully played unintentionally paused ${kind} element`);
            }).catch(err => {
              log.warn(`error playing paused ${kind} element`, err);
            });
          }
        });
```